fitIn/SignUpViewController.swift

```diff
@@ -14,56 +14,126 @@ import AWSDynamoDB
+                inputValidationConditions[1] = true
+            }
+            else
+            {
+                AgeTextField.textColor = UIColor.red
+                CreateNewUserVariable.backgroundColor = UIColor(red: 204/255, green: 17/255, blue: 0/255, alpha: 1.0)
+                inputValidationConditions[1] = false
+            }
+        }
+        
+        if (EmailTextField.text!.count > 0)
+        {
+            if (currentUser?.updateEmailAddress(EmailTextField.text!) == true)
+            {
+                //userEditEmailAddressField must be more than 5 characters long, and must contain a "@" character
+                //input has been validated at this point, but the constraints can be modified in the future
+                EmailTextField.textColor = UIColor.black
+                inputValidationConditions[2] = true
+            }
+            else
+            {
+                //input either is not long enough, or does not contain a "@" character
+                //userEditProfileSaveChangesButton.setTitleColor(UIColor.red, for: UIControlState.normal)
+                EmailTextField.textColor = UIColor.red
+                CreateNewUserVariable.backgroundColor = UIColor(red: 204/255, green: 17/255, blue: 0/255, alpha: 1.0)
+                inputValidationConditions[2] = false
+            }
         }
-        else
+        
+        if (PasswordTextField.text!.count > 0 && ConfirmPasswordTextField.text!.count > 0  )
         {
-            //5 years old is the arbitrary default age
-            UserProfile.current()?.userAge = 5
+            if (PasswordTextField.text!.count > 3 && ConfirmPasswordTextField.text!.count > 3 && PasswordTextField.text! == ConfirmPasswordTextField.text!)
+            {
+                currentUser?.passwordToken = PasswordTextField.text!
+                PasswordTextField.textColor = UIColor.black
+                ConfirmPasswordTextField.textColor = UIColor.black
+                inputValidationConditions[3] = true
+                
+            }
+            else
+            {
+                PasswordTextField.textColor = UIColor.red
+                ConfirmPasswordTextField.textColor = UIColor.red
+                CreateNewUserVariable.backgroundColor = UIColor(red: 204/255, green: 17/255, blue: 0/255, alpha: 1.0)
+                inputValidationConditions[3] = false
+            }
+            
+        }
+        
+        if (inputValidationConditions[0] == true && inputValidationConditions[1] == true && inputValidationConditions[2] == true && inputValidationConditions[3] == true && inputValidationConditions[4] == true && inputValidationConditions[5] == true) {
+            //userEditProfileSaveChangesButton.setTitleColor(UIColor.green, for: UIControlState.normal)
+            CreateNewUserVariable.backgroundColor = UIColor(red: 0/255, green: 155/255, blue: 77/255, alpha: 1.0)
+            dynamoHandler.putUserProfile(currentUser!)
+            currentUser?.isUserLoggedIn = true
```

Comment:
This should probably be moved to inside the db where it says "success, new user"